### PR TITLE
ASAN/UBSAN fixes

### PR DESCRIPTION
**Pull request by jengelh**
## Pull Request Type

- [x] Runtime changes
  - [x] Other changes

### Description

lesser UBSAN fixes

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

File: lib/grdefs.h [hunk 1]
```diff
@@ -50,6 +50,7 @@
 #ifndef GRDEFS_H
 #define GRDEFS_H
 
+#include <cstdint>
 #include "pstypes.h"
```
Comment (winterheart) [inline lib/grdefs.h:54]: Including `cstdint` usually makes `pstypes.h` redundant.

File: lib/grdefs.h [hunk 2]
```diff
@@ -82,9 +83,9 @@ static const ddgr_color GR_NULL = 0xffffffff, // don't do a thing with this.
 #define GR_COLOR_CHAR 1 // ASCII 1 and (r,g,b) changes current text color in string.
 
 //	MACROS
-static inline ddgr_color GR_RGB(int r, int g, int b) { return ((r << 16) + (g << 8) + b); }
+static inline ddgr_color GR_RGB(uint8_t r, uint8_t g, uint8_t b) { return ((r << 16) + (g << 8) + b); }
 
-static inline uint16_t GR_RGB16(int r, int g, int b) { return (((r >> 3) << 10) + ((g >> 3) << 5) + (b >> 3)); }
+static inline uint16_t GR_RGB16(uint8_t r, uint8_t g, uint8_t b) { return (((r >> 3) << 10) + ((g >> 3) << 5) + (b >> 3)); }
```
Comment (winterheart) [inline lib/grdefs.h:88]: Probably is better change r,g,b to `uint32_t`? All calls defines r,g,b as `int`.

Comment (winterheart) [reply lib/grdefs.h:87]: Is there any changes are planned?

Comment (jengelh) [reply lib/grdefs.h:87]: (Not sure where Github lost my comment, but here goes again: )

Changing to `uint32_t` is a *big* problem. The callsite effectively uses `char b = 255; GR_RGB16(r,g,b);`, which causes b=0xffffffff, and all those blue bits spill over into `r` and `g`, making the result of GR_RGB useless.
So no change planned, uint8_t is just right. (A function probably should not take types whose range is not handled anyway.)

**Comment by winterheart** [comment]
Tested on Linux, no issues found.

**Pull request merged**